control-plane/pkg/reconciler/trigger/controller.go

```diff
@@ -153,7 +156,8 @@ func NewController(ctx context.Context, watcher configmap.Watcher, configs *conf
 	secretinformer.Get(ctx).Informer().AddEventHandler(controller.HandleAll(reconciler.Tracker.OnChanged))
 
 	// Reconciler Trigger when the OIDC service account changes
-	serviceaccountInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+	// serviceaccountInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		oidcServiceaccountInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
 		FilterFunc: controller.FilterController(&eventing.Trigger{}),
 		Handler:    controller.HandleAll(impl.EnqueueControllerOf),
```

Comment:
@creydr we need to make sure that the trigger belogs to a broker with "our" broker class

Reply:
@karkabbage you can check on https://github.com/knative/eventing/pull/7849, how this was done in eventing-core

Reply:
Addressed in 3e9129d0e60dd92de599837b8b80ee66bba772e6